In [1]:
import pandas as pd
import numpy as np


# Migraciones

In [2]:
cod_paises = pd.read_csv('Datasets/Metadata_Country_API_SM.POP.NETM_DS2_es_csv_v2_5368209.csv')
migracion_neta_por_anio = pd.read_csv('Datasets/API_SM.POP.NETM_DS2_es_csv_v2_5368209.csv',skiprows=4)

In [4]:
cod_paises = cod_paises.iloc[:,:-1]
cod_paises

Country Name Country Code  \
0              Aruba          ABW   
1                NaN          AFE   
2         Afganistán          AFG   
3                NaN          AFW   
4             Angola          AGO   
..               ...          ...   
261           Kosovo          XKX   
262  Yemen, Rep. del          YEM   
263        Sudáfrica          ZAF   
264           Zambia          ZMB   
265         Zimbabwe          ZWE   

                                                Region  
0                                                  NaN  
1                                                  NaN  
2                                      Asia meridional  
3                                                  NaN  
4    África al sur del Sahara (excluido altos ingre...  
..                                                 ...  
261    Europa y Asia central (excluido altos ingresos)  
262  Oriente Medio y Norte de África (excluido alto...  
263  África al sur del Sahara (excluido altos ingre...  
264  África al sur del Sahara (excluido altos ingre...  
265  África al sur del Sahara (excluido altos ingre...  

[266 rows x 3 columns]

In [5]:
cod_paises.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 266 entries, 0 to 265
Data columns (total 3 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Country Name  264 non-null    object
 1   Country Code  266 non-null    object
 2   Region        136 non-null    object
dtypes: object(3)
memory usage: 6.4+ KB


In [6]:
cod_paises.duplicated().unique()

array([False])

* columnas de relacion:
    - Country Code
    - Country Name

* tipos de datos:
    - catergoticos:
        - nominal:
            - Country Name	
            - Country Code
            - Region	
            - Income_Group

* Datos Faltantes: 
    - Hay valores faltantes en :
        - Country Name(2)
        - Region(130)

    - Propuesta: llenar a mano los datos faltantes. 

* Sin datos duplicados


    

* redusco el dataset de migracionse a el rango de 2000 a 2021

In [7]:
migracion_neta_por_anio = pd.concat([migracion_neta_por_anio.iloc[:,:4],migracion_neta_por_anio.iloc[:,-23:-1]],axis=1)

In [8]:
migracion_neta_por_anio

Country Name Country Code  Indicator Name Indicator Code       2000  \
0              Aruba          ABW  Migración neta    SM.POP.NETM     1235.0   
1                NaN          AFE  Migración neta    SM.POP.NETM  -564510.0   
2         Afganistán          AFG  Migración neta    SM.POP.NETM -1007135.0   
3                NaN          AFW  Migración neta    SM.POP.NETM   -22651.0   
4             Angola          AGO  Migración neta    SM.POP.NETM    53527.0   
..               ...          ...             ...            ...        ...   
261           Kosovo          XKX  Migración neta    SM.POP.NETM    -9176.0   
262  Yemen, Rep. del          YEM  Migración neta    SM.POP.NETM   -87753.0   
263        Sudáfrica          ZAF  Migración neta    SM.POP.NETM   -80388.0   
264           Zambia          ZMB  Migración neta    SM.POP.NETM    -5152.0   
265         Zimbabwe          ZWE  Migración neta    SM.POP.NETM  -137321.0   

         2001       2002      2003      2004      2005  ...      2012  \
0       560.0      350.0     324.0     315.0     579.0  ...     229.0   
1   -409728.0  -452301.0 -342557.0 -331713.0 -357877.0  ... -285231.0   
2   -192286.0  1327074.0  388632.0 -248616.0  252185.0  ...  105905.0   
3    -38737.0   -39042.0  -99836.0  -41511.0 -137920.0  ... -435876.0   
4     56959.0    60505.0   59438.0   66585.0   64009.0  ...   96056.0   
..        ...        ...       ...       ...       ...  ...       ...   
261   -1995.0   -29565.0  -27077.0  -30715.0  -33861.0  ...  -15323.0   
262  -93742.0   -99706.0 -104794.0 -102701.0  -51360.0  ...  -20983.0   
263  -29826.0    50323.0   65721.0   58977.0   31424.0  ...   68846.0   
264   23832.0     3209.0   20873.0    7828.0   35351.0  ...    9009.0   
265 -136966.0  -147110.0 -118634.0 -156566.0 -164732.0  ...  -66709.0   

         2013      2014      2015      2016      2017      2018      2019  \
0       134.0      88.0     177.0     170.0     218.0     367.0     412.0   
1    -92842.0 -199608.0  394925.0 -864708.0 -343075.0 -366105.0 -187410.0   
2     48076.0  255611.0 -281739.0  -90238.0  -47090.0  -47205.0   -8082.0   
3   -377305.0 -280418.0 -360393.0 -167286.0 -200823.0 -255888.0 -328552.0   
4     91528.0   84532.0   74170.0   91623.0   77733.0   61632.0   58986.0   
..        ...       ...       ...       ...       ...       ...       ...   
261  -40468.0  -56637.0  -14819.0  -31383.0  -33495.0  -34137.0  -33469.0   
262  -23747.0  -30048.0  -38355.0  -39688.0  -39672.0  -39699.0  -39769.0   
263   63794.0  290874.0  644996.0 -866101.0  -10681.0   27265.0   22728.0   
264   15574.0   16580.0   16283.0   17285.0   20268.0   18019.0   18019.0   
265  -59918.0  -59918.0  -59918.0  -59918.0  -59918.0  -59918.0  -59918.0   

         2020      2021  
0         0.0     501.0  
1    -48955.0 -179444.0  
2    166821.0 -183672.0  
3    -99761.0 -264802.0  
4      7557.0   29089.0  
..        ...       ...  
261  -15050.0  -15050.0  
262  -86100.0 -101468.0  
263   30852.0   10934.0  
264    9015.0    9015.0  
265  -29955.0  -25005.0  

[266 rows x 26 columns]

In [9]:
migracion_neta_por_anio.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 266 entries, 0 to 265
Data columns (total 26 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Country Name    264 non-null    object 
 1   Country Code    266 non-null    object 
 2   Indicator Name  266 non-null    object 
 3   Indicator Code  266 non-null    object 
 4   2000            265 non-null    float64
 5   2001            265 non-null    float64
 6   2002            265 non-null    float64
 7   2003            265 non-null    float64
 8   2004            265 non-null    float64
 9   2005            265 non-null    float64
 10  2006            265 non-null    float64
 11  2007            265 non-null    float64
 12  2008            265 non-null    float64
 13  2009            265 non-null    float64
 14  2010            265 non-null    float64
 15  2011            265 non-null    float64
 16  2012            265 non-null    float64
 17  2013            265 non-null    flo

In [10]:
migracion_neta_por_anio.duplicated().unique()

array([False])

In [11]:
migracion_neta_por_anio['Indicator Name'].unique()
migracion_neta_por_anio['Indicator Code'].unique()

array(['SM.POP.NETM'], dtype=object)

* columnas de relacion: 
    - Country Name	
    - Country Code	
* tipos de datos:
    - catergoticos:
        - nominal:
            - Indicator Name	
            - Indicator Code	
            
    - cuantitativos:
        - discretos:
            - columnas de 2000 a 2021(1)

* Datos Faltantes: 
    - Hay valores faltantes en :
        - columnas de 2000 a 2021(1)

* Observaciones:
    - La fila de "No clasificado" con codigo de pais "INX" no posee datos para ningun anio.
    - Indicator Name e Indicator code tienen un unico valor
    
* Propuestas:
    - para los valores faltantes de Country Name: agregarlos a mano o usando la tabla de cod_paises
    - para la fija de 'No clasificado': eliminarla pues no hay datos ni forma de completar la

# Remesas

In [13]:
remesas_entrantes= pd.read_excel('remesas/inward_remittance_flows_as_of_dec._2_2022_0.xlsx',decimal='.',)
remesas_salientes= pd.read_excel('remesas/outward_remittance_flows_brief_37_as_of_nov28_2022_2_1.xlsx',skiprows=1)

### Remesas Entrantes

* Saco las 10 ulitmas filas pues son observaciones, y limito el rango de anio a 2000 a 2022

In [14]:
remesas_entrantes = pd.concat([remesas_entrantes.iloc[:,0],remesas_entrantes.iloc[:,-24:-1]],axis=1)
remesas_entrantes = remesas_entrantes.iloc[:-10]


In [15]:
remesas_entrantes.head()

Migrant remittance inflows (US$ million)   2000   2001    2002        2003  \
0                              Afghanistan    0.0    0.0    0.00    0.000000   
1                                  Albania  597.8  699.3  733.57  888.748582   
2                                  Algeria    0.0    0.0    0.00    0.000000   
3                           American Samoa    NaN    NaN     NaN         NaN   
4                                  Andorra    0.0    0.0    0.00    0.000000   

          2004         2005         2006       2007        2008  ...  \
0     0.000000     0.000000            0          0        89.5  ...   
1  1160.672105  1289.704316  1359.467325    1468.02      1865.6  ...   
2     0.000000   170.000000        189.0  99.004563  103.631887  ...   
3          NaN          NaN          NaN        NaN         NaN  ...   
4     0.000000          NaN          NaN        NaN         NaN  ...   

          2013         2014         2015         2016         2017  \
0   347.165292   253.367822   348.624717   627.710802   822.731630   
1  1281.848114  1421.007454  1290.863508  1306.009167  1311.822432   
2   209.601443  2452.442617  1997.393458  1989.023597  1791.887073   
3          NaN          NaN          NaN          NaN          NaN   
4          NaN          NaN          NaN          NaN          NaN   

          2018         2019         2020         2021        2022e  
0   803.546454   828.571904   788.917115          300   350.000000  
1  1458.210056  1472.812242  1465.987212  1718.320554  1800.000000  
2  1984.998399  1785.838683  1699.608935  1759.095247  1829.459057  
3          NaN          NaN          NaN          NaN          NaN  
4          NaN    21.100000    47.416324          NaN          NaN  

[5 rows x 24 columns]

In [16]:
remesas_entrantes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 216 entries, 0 to 215
Data columns (total 24 columns):
 #   Column                                    Non-Null Count  Dtype  
---  ------                                    --------------  -----  
 0   Migrant remittance inflows (US$ million)  216 non-null    object 
 1   2000                                      202 non-null    float64
 2   2001                                      202 non-null    float64
 3   2002                                      202 non-null    float64
 4   2003                                      202 non-null    float64
 5   2004                                      202 non-null    float64
 6   2005                                      195 non-null    float64
 7   2006                                      196 non-null    object 
 8   2007                                      196 non-null    object 
 9   2008                                      196 non-null    object 
 10  2009                                  

In [17]:
remesas_entrantes.duplicated().unique()

array([False])

* columnas de relacion: 

* tipos de datos:
    - catergoticos:
        - nominal:
            - Migrant remittance inflows (US$ million)
            
    - cuantitativos:
        - continuos:
            - columnas de la segunda a la ultiama

* Datos Faltantes: 
    - Hay valores faltantes en :
        - todas exepto la primera
* Observaciones:
    - solo hay datos de 215 paices y un total del mundo 
    - hay datos hasta el 2022

* Propuesta:
    - ante datos nulos llenar con ceros.
    - cambiar el monbre dela pirmera columna a Country o Pais

    

### Remesas Saliente

* Saco las 10 ulitmas filas pues son observaciones, y limito el rango de anio a 2000 a 2022

* limito el rango de 2000 a 2021, y saco las 10 ultimas columnas del excel que son informaicion.

In [18]:
remesas_salientes = pd.concat([remesas_salientes.iloc[:,0],remesas_salientes.iloc[:-10,11:-49]],axis=1)
remesas_salientes = remesas_salientes.iloc[:-11]


In [19]:
remesas_salientes

Outward remittance flows (US$ million)  2000       2001       2002  \
0                              Afghanistan   NaN        NaN        NaN   
1                                  Albania   0.0   0.000000   0.000000   
2                                  Algeria   NaN        NaN        NaN   
3                           American Samoa   NaN        NaN        NaN   
4                                  Andorra   NaN        NaN        NaN   
..                                     ...   ...        ...        ...   
209                  Virgin Islands (U.S.)   NaN        NaN        NaN   
210                     West Bank and Gaza   5.6   4.900869   5.095333   
211                            Yemen, Rep.   NaN        NaN        NaN   
212                                 Zambia  24.1  28.400000  37.500000   
213                               Zimbabwe   0.0   0.000000   0.000000   

          2003       2004        2005        2006        2007        2008  \
0          NaN        NaN         NaN         NaN         NaN  216.600000   
1     4.135728   4.863740         NaN   26.532132    9.940000  279.700000   
2          NaN        NaN   27.000000   35.000000   48.863733   26.782545   
3          NaN        NaN         NaN         NaN         NaN         NaN   
4          NaN        NaN         NaN         NaN         NaN         NaN   
..         ...        ...         ...         ...         ...         ...   
209        NaN        NaN         NaN         NaN         NaN         NaN   
210  17.883733  10.760551    6.475643    6.788488    7.929239    8.299895   
211        NaN        NaN  109.495910  120.440000  318.720000  336.840000   
212  71.500000  76.100000   93.700000  115.457000  123.930000  138.885000   
213   0.000000   0.000000    0.000000    0.000000    0.000000    0.000000   

     ...        2012        2013        2014        2015        2016  \
0    ...  323.642538  444.730790  524.163479  228.991809  167.894464   
1    ...  215.775644  190.551294  178.722842  153.312400  147.156757   
2    ...   44.242619   38.715810  295.922112   72.128991   76.633496   
3    ...         NaN         NaN         NaN         NaN         NaN   
4    ...         NaN         NaN         NaN         NaN         NaN   
..   ...         ...         ...         ...         ...         ...   
209  ...         NaN         NaN         NaN         NaN         NaN   
210  ...   48.109831   52.642111   36.128914   30.398350   30.393004   
211  ...  337.707048  333.295533  335.395533  333.389518  332.700000   
212  ...   96.922000   77.324471   81.199881   72.104699   63.405836   
213  ...    9.889461   10.354266   10.840916   11.350439   13.844871   

           2017        2018        2019        2020        2021  
0    143.979106  234.618701  217.292318  225.420606         NaN  
1    106.330983  114.879337  119.679641  124.630376   140.34431  
2    214.378057   87.006383   81.545365  149.237304   81.225032  
3           NaN         NaN         NaN         NaN         NaN  
4           NaN         NaN   91.500000   83.804275         NaN  
..          ...         ...         ...         ...         ...  
209         NaN         NaN         NaN         NaN         NaN  
210   27.462198   22.347335   29.079904   21.881177    9.330391  
211         NaN         NaN         NaN         NaN         NaN  
212  121.370621  132.001364  103.806724   88.510886  106.780374  
213   18.575000   29.278892   13.480566   21.707094         NaN  

[214 rows x 23 columns]

In [20]:
remesas_salientes['Outward remittance flows (US$ million)'].unique()

array(['Afghanistan', 'Albania', 'Algeria', 'American Samoa', 'Andorra',
       'Angola', 'Antigua and Barbuda', 'Argentina', 'Armenia', 'Aruba',
       'Australia', 'Austria', 'Azerbaijan', 'Bahamas, The', 'Bahrain',
       'Bangladesh', 'Barbados', 'Belarus', 'Belgium', 'Belize', 'Benin',
       'Bermuda', 'Bhutan', 'Bolivia', 'Bosnia and Herzegovina',
       'Botswana', 'Brazil', 'Brunei Darussalam', 'Bulgaria',
       'Burkina Faso', 'Burundi', 'Cabo Verde', 'Cambodia', 'Cameroon',
       'Canada', 'Cayman Islands', 'Central African Republic', 'Chad',
       'Channel Islands', 'Chile', 'China', 'Colombia', 'Comoros',
       'Congo, Dem. Rep.', 'Congo, Rep.', 'Costa Rica', "Cote d'Ivoire",
       'Croatia', 'Cuba', 'Curacao', 'Cyprus', 'Czech Republic',
       'Denmark', 'Djibouti', 'Dominica', 'Dominican Republic', 'Ecuador',
       'Egypt, Arab Rep.', 'El Salvador', 'Equatorial Guinea', 'Eritrea',
       'Estonia', 'Eswatini', 'Ethiopia', 'Faeroe Islands', 'Fiji',
       'Finland'

In [21]:
remesas_salientes.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 214 entries, 0 to 213
Data columns (total 23 columns):
 #   Column                                  Non-Null Count  Dtype  
---  ------                                  --------------  -----  
 0   Outward remittance flows (US$ million)  214 non-null    object 
 1   2000                                    153 non-null    float64
 2   2001                                    154 non-null    float64
 3   2002                                    162 non-null    float64
 4   2003                                    163 non-null    float64
 5   2004                                    165 non-null    float64
 6   2005                                    174 non-null    float64
 7   2006                                    177 non-null    float64
 8   2007                                    180 non-null    float64
 9   2008                                    181 non-null    float64
 10  2009                                    183 non-null    float6

In [22]:
remesas_salientes.duplicated().unique()

array([False])

* columnas de relacion: 

* tipos de datos:
    - catergoticos:
        - nominal:
            - Outward remittance flows (US$ million)
            
    - cuantitativos:
        - continuos:
            - columnas de la segunda a la ultiama

* Datos Faltantes: 
    - Hay valores faltantes en :
        - la unica completa es la primer columna

* Duplicados: niguno
* Observaciones:
    - solo hay datos de 214 paices
    - solo hay datos hasta el 2021 

* Propuesta:
    - ante datos continuos nulos llenar con ceros.
    - cambiar el monbre de la pirmera columna a Country o Pais

    